### PR TITLE
Add `static_checker` to parser utils

### PR DESCRIPTION
**Pull request by GlassOfWhiskey**
This commit partially ports the `static_checker` logic from `cwltool` into the `cwl-utils` library, using Schema SALAD objects instead of Python dictionaries.

### CODE REVIEW

File: tests/test_parser_utils.py [hunk 1]
```diff
@@ -13,12 +14,242 @@
...
+import cwl_utils.parser.utils
 from cwl_utils.errors import WorkflowException
 from cwl_utils.parser import load_document_by_uri
 
 from .util import get_data
 
 
+def test_static_checker_fail() -> None:
+    """Confirm that static type checker raises expected exception."""
+    with pytest.raises(ValidationException):
```
Comment (mr-c) [inline tests/test_parser_utils.py:26]: Could these checks be made more specific about the expected exception message?

https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

File: tests/test_parser_utils.py [hunk 2]
```diff
@@ -13,12 +14,242 @@
...
+            Path(get_data("testdata/cond-single-source-wf-004.1.cwl"))
+            .resolve()
+            .as_uri()
+        )
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+
+def test_static_checker_success() -> None:
+    """Confirm that static type checker correctly validates workflows."""
```
Comment (mr-c) [inline tests/test_parser_utils.py:103]: Can you parameterize these so the test is not so repetitive?

https://docs.pytest.org/en/6.2.x/parametrize.html

**Review by mr-c — CHANGES_REQUESTED** [review]
So exciting! A few requests about the tests

File: tests/test_parser_utils.py (hunk 2 above)
Comment (GlassOfWhiskey) [reply tests/test_parser_utils.py]: I did it

File: tests/test_parser_utils.py (hunk 1 above)
Comment (GlassOfWhiskey) [reply tests/test_parser_utils.py]: Did it

**Pull request merged**